src/lib.rs

```diff
@@ -135,6 +135,10 @@ pub use secp256k1_sys as ffi;
 #[cfg(any(test, feature = "rand"))] use rand::Rng;
 #[cfg(any(test, feature = "std"))] extern crate core;
 #[cfg(all(test, target_arch = "wasm32"))] extern crate wasm_bindgen_test;
+#[cfg(feature = "alloc")] extern crate alloc;
+
+#[cfg(all(feature = "std", feature = "alloc"))]
+compile_error!("std and alloc are mutually exclusive");
```

Comment:
I've done the same in another PR https://github.com/rust-bitcoin/bitcoin_hashes/pull/127

however, this causes an issue with how current resolver threat dependencies, taking the union of features if different crates specify different features. New resolver fixes this issue https://blog.rust-lang.org/2021/03/25/Rust-1.51.0.html#cargos-new-feature-resolver but it's much more modern than our MSRV

Reply:
so are you thinking that if they are both set, then we should act as if `std` only is set?

Reply:
added a fixup for this